oioioi/oireports/views.py

```diff
@@ -167,7 +167,6 @@ def _serialize_report(user, problem_instances, test_groups):
                 codefile=source_file.file.name,
             )
         )
-        source_file.close()
```

Comment:
I think it'd be worth double-checking if:
- generating PDF/XML reports works without this
- this doesn't cause an FD leak

Reply:
will check later

Reply:
when reverting this change generating PDF report throws
```
AttributeError: '_FileLikeFromResponse' object has no attribute 'close'
```

Reply:
It would make more sense to have `source_file.file.close()`. After digging on the issue, django model's File can open itself implicitly when reading. It should be closed only it it was explicitly opened. I decided not to put more time into it now, as it seems to be fine and unfortunately other commits require much more time and work.